### PR TITLE
Migrate to pkg:lints, fix related lints, cleanup in pubspec

### PR DESCRIPTION
**Pull request by kevmoo**
Removed pre-release versions from changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,24 +1,9 @@
+# 0.10.11-dev
+
 ## 0.10.10
 
 * Stable release for null safety.
 
-## 0.10.10-nullsafety.3
```
Comment (jakemac53) [inline CHANGELOG.md:5]: @natebosch are we generally removing these once stable is released? Given these are real versions on pub I am not sure why we would remove them?

File: lib/printer.dart [hunk 2]
```diff
@@ -10,8 +10,8 @@ import 'package:source_span/source_span.dart';
 import 'builder.dart';
 import 'src/source_map_span.dart';
 
-const int _LF = 10;
-const int _CR = 13;
+const int lineFeed = 10;
```
Comment (jakemac53) [inline lib/printer.dart]: You made these public, was that intentional?

Comment (kevmoo) [reply lib/printer.dart]: Gah! no! Oops

Comment (kevmoo) [reply lib/printer.dart]: I mean, I wanted to share them, but not make the API public

File: CHANGELOG.md (hunk 1 above)
Comment (kevmoo) [reply CHANGELOG.md:5]: They are generally hidden from users. At this point, a user would have to explicitly opt-in to these. I feel they are better hidden.

File: lib/printer.dart (hunk 2 above)
Comment (kevmoo) [reply lib/printer.dart]: fixed!

File: CHANGELOG.md (hunk 1 above)
Comment (natebosch) [reply CHANGELOG.md:5]: Yeah, we started collapsing these logs down to just the stable ones very late into the process when we were publishing for stable null safety. For the packages where we hadn't done originally that we've been going back and collapsing it. It's useful in the github history, but I don't think it's worth the noise to keep it in the file.

**Review by natebosch — APPROVED** [review]

**Review by jakemac53 — APPROVED** [review]

**Pull request merged**